applications/CompressiblePotentialFlowApplication/tests/cpp_tests/test_potential_flow_utilities.cpp

```diff
@@ -163,5 +163,142 @@ KRATOS_TEST_CASE_IN_SUITE(ComputePerturbationLocalMachNumber, CompressiblePotent
+
+    double reference_derivative = 1.16201001910861e-05;
+
+    KRATOS_CHECK_NEAR(mach_derivative, reference_derivative, 1e-16);
+}
+
+// Checks the function ComputeLocalMachSquaredDerivative from the utilities, supersonic local Mach number
+KRATOS_TEST_CASE_IN_SUITE(ComputeLocalMachSquaredDerivativeSupersonicMach, CompressiblePotentialApplicationFastSuite) {
+    Model this_model;
+    ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+    model_part.GetProcessInfo()[FREE_STREAM_DENSITY] = 1.225;
+    model_part.GetProcessInfo()[FREE_STREAM_MACH] = 0.6;
+    model_part.GetProcessInfo()[HEAT_CAPACITY_RATIO] = 1.4;
+    model_part.GetProcessInfo()[SOUND_VELOCITY] = 340.0;
+    model_part.GetProcessInfo()[MACH_SQUARED_LIMIT] = 3.0;
+
+    BoundedVector<double, 3> free_stream_velocity = ZeroVector(3);
+    free_stream_velocity(0) = model_part.GetProcessInfo().GetValue(FREE_STREAM_MACH) *
+                              model_part.GetProcessInfo().GetValue(SOUND_VELOCITY);
+    model_part.GetProcessInfo()[FREE_STREAM_VELOCITY] = free_stream_velocity;
+
+    array_1d<double, 2> velocity(2, 0.0);
+    velocity[0] = 272.0 * sqrt(134.0/21.0);
+
+    double sq_local_mach_number = PotentialFlowUtilities::ComputeLocalMachNumberSquared<2, 3>(velocity, model_part.GetProcessInfo());
+    double local_mach_number = sqrt(sq_local_mach_number);
+
+    double mach_derivative = PotentialFlowUtilities::ComputeLocalMachSquaredDerivative<2, 3>(velocity,
+                local_mach_number, model_part.GetProcessInfo());
+                
+    double reference_derivative = 2.06579558952642e-05;
+
+    KRATOS_CHECK_NEAR(mach_derivative, reference_derivative, 1e-16);
+}
+
+// Checks the function ComputeMaximumVelocitySquared from the utilities
+KRATOS_TEST_CASE_IN_SUITE(ComputeMaximumVelocitySquared, CompressiblePotentialApplicationFastSuite) {
+    Model this_model;
+    ModelPart& model_part = this_model.CreateModelPart("Main", 3);
+
+    model_part.GetProcessInfo()[FREE_STREAM_DENSITY] = 1.225;
+    model_part.GetProcessInfo()[FREE_STREAM_MACH] = 0.6;
+    model_part.GetProcessInfo()[HEAT_CAPACITY_RATIO] = 1.4;
+    model_part.GetProcessInfo()[SOUND_VELOCITY] = 340.0;
+    model_part.GetProcessInfo()[MACH_SQUARED_LIMIT] = 3.0;
+
+    BoundedVector<double, 3> free_stream_velocity = ZeroVector(3);
+    free_stream_velocity(0) = model_part.GetProcessInfo().GetValue(FREE_STREAM_MACH) *
+                              model_part.GetProcessInfo().GetValue(SOUND_VELOCITY);
+    model_part.GetProcessInfo()[FREE_STREAM_VELOCITY] = free_stream_velocity;
+
+    // Max local Mach number = sqrt(3.0), hard coded in ComputeMaximumVelocitySquared
+
+    double reference_sq_max_velocity = 232356.0000000000000000;
+
+    double sq_max_velocity = PotentialFlowUtilities::ComputeMaximumVelocitySquared<2, 3>(model_part.GetProcessInfo());
+
+    KRATOS_CHECK_NEAR(sq_max_velocity, reference_sq_max_velocity, 1e-10);
```

Comment:
KRATOS_CHECK_RELATIVE_NEAR

Reply:
@inigolcanalejo Also had issues using CHECK_RELATIVE_NEAR.

Reply:
ok lets talk about it tomorrow